featuretools/primitives/rolling_primitive_utils.py

```diff
@@ -136,14 +123,14 @@ def apply_roll_with_offset_gap(window, gap_offset, reducer_fn, min_periods):
     return reducer_fn(window)
 
 
-def _check_window_size(window_size):
+def _check_window_size(window_size: Union[int, str]) -> None:
```

Comment:
I'm wondering if were's a custom type we could create for this `Union[int, str])` (which shows up 7 times in this PR) to encompass the idea of it being either an offset string or number of rows.

Reply:
Personally, I like to avoid custom typing unless the type gets pretty crazy. Because python is dynamically typed having a simple union type like this is fine. Also once we eventually introduce mypy or pyright into the repo, I think having a simpler type will be much easier to deal with. I'm open to other opinions though.

Reply:
That makes sense to me!